tests/TestHelpers/HttpRequestHelper.php

```diff
@@ -216,19 +218,22 @@ public static function sendRequest(
 				$client
 			);
 
-			if ($response->getStatusCode() >= 400 && $response->getStatusCode() !== self::HTTP_TOO_EARLY) {
+			if ($response->getStatusCode() >= 400 && $response->getStatusCode() !== self::HTTP_TOO_EARLY && $response->getStatusCode() !== self::HTTP_CONFLICT) {
 				$sendExceptionHappened = true;
 			}
 
 			if ($debugResponses) {
 				self::debugResponse($response);
 			}
 			$sendCount = $sendCount + 1;
-			$loopAgain = !$sendExceptionHappened && ($response->getStatusCode() === self::HTTP_TOO_EARLY) && ($sendCount <= $sendRetryLimit);
+			$loopAgain = !$sendExceptionHappened && ($response->getStatusCode() === self::HTTP_TOO_EARLY ||
+						($response->getStatusCode() === self::HTTP_CONFLICT && $isGivenStep)) &&
+						$sendCount <= $sendRetryLimit;
```

Comment:
can you add some comments regarding the early fail and conflict status code. It might be confusing when someone looks at it. You can add comment what is the reason of doing this for both.